src/options.cpp

```diff
@@ -138,6 +144,46 @@ void global_settings_via_options::set_oauth_10_support(const po::variables_map &
   }
 }
 
+void global_settings_via_options::set_ratelimiter_ratelimit(const po::variables_map &options) {
+  if (options.count("ratelimit")) {
+    auto parsed_bytes_per_sec = options["ratelimit"].as<long>();
+    if (parsed_bytes_per_sec < 0)
+      throw std::invalid_argument("ratelimit must be a positive number");
+    if (parsed_bytes_per_sec > 1024 * 1024 * 1024)
+      throw std::invalid_argument("ratelimit must be 1GB or less.");
+    m_ratelimiter_ratelimit = parsed_bytes_per_sec;
+  }
+
+  if (options.count("moderator-ratelimit")) {
+    auto parsed_bytes_per_sec = options["moderator-ratelimit"].as<long>();
+    if (parsed_bytes_per_sec < 0)
+      throw std::invalid_argument("moderator-ratelimit must be a positive number");
+    if (parsed_bytes_per_sec > 1024 * 1024 * 1024)
+      throw std::invalid_argument("moderator-ratelimit must be 1GB or less.");
+    m_moderator_ratelimiter_ratelimit = parsed_bytes_per_sec;
+  }
+}
+
+void global_settings_via_options::set_ratelimiter_maxdebt(const po::variables_map &options) {
+ if (options.count("maxdebt")) {
+    auto parsed_max_bytes = options["maxdebt"].as<long>();
+    if (parsed_max_bytes < 0)
+      throw std::invalid_argument("maxdebt must be a positive number");
+    if (parsed_max_bytes > 3500)
+      throw std::invalid_argument("maxdebt must be 3500 MB or less.");
```

Comment:
This could confuse some folks, since it looks similar to the error message for ratelimit. However, we're actually looking for a value <= 3500.

Maybe: `maxdebt must be 3500 or less (unit: MB)`.

Reply:
That's pretty much the only thing I still have on my list. I would continue with merging afterwards. Hopefully we could give it a try on the dev instance.

Reply:
Updated

Reply:
Great, thanks a lot! I will merge right after the two pipeline checks are done.